cis/processor.py

```diff
@@ -0,0 +1,127 @@
+"""Iterative Stream Processor plugin for handling steps during data storage."""
+import json
+import logging
+import os
+
+from cis import user
+
+from cis.libs import encryption
+from cis.libs import streams
+from cis.libs import validation
+
+logger = logging.getLogger(__name__)
+
+class OperationDelegate(object):
+    def __init__(self, boto_session=None, publisher=None, signature=None, encrypted_profile_data=None):
+        self.boto_session = boto_session
+        self.dry_run = True
+        self.decryptor = encryption.Operation(boto_session=boto_session)
+        self.encrypted_profile_data = encrypted_profile_data
+        self.kinesis_client = None
+        self.publisher = publisher
+        self.signature = signature
+        self.stage = self._get_stage()
+        self.user = None
+
+    def run(self):
+        # Determine what stage of processing we are in and call the corresponding functions.
+
+        self.decrytped_profile = json.loads(self._decrypt_profile_packet())
+
+        if self.stage == 'validator':
```

Comment:
Again I would avoid checking against strings and have all these nested ``if/elif/else`` cases which are very error prone and hard to follow. Since you are going the OO way it would be far more elegant to subclass the different stages.